### PR TITLE
Set parent relationship only if they have the same type

### LINKED ISSUE #16: Sync is crashing when syncing children JSON to Object with not existing relationship
I have a Model with many Relationship. In the JSON I have a simpler model without all relationship

When I Sync numbers, it crashes because trying to set Parent `unit` property on Number (unit is relationship to OrganizationUnit, but I don't want to sync organisations here, just numbers) 

Model Example :
screen shot 2015-03-03 at 09 12 52

JSON Example:
``` json
[{
  "id": 1,
  "number": 10.0,
  "numberType": 0,
  "children" : [
    {
      "id": 2,
      "number": 11.0,
      "numberType": 0
    },
    {
      "id": 3,
      "number": 12.0,
      "numberType": 0,
    }
  ]
}]
```

*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Unacceptable type of value for to-one relationship: property = "unit"; desired type = REMAOrganizationUnit; given type = REMANumber; value = <REMANumber: 0x7aeac650> 

It crash here :)
screen shot 2015-03-03 at 10 11 21

it has a relationship with name `unit` that point to parent organization, but there is no organization in the JOSN and I don't want to process it

### PR DESCRIPTION
**Pull request by kostiakoval**
Fix #16 
Also it would nice to add test for that case

### CODE REVIEW

File: Source/Sync.m [hunk 1]
```diff
@@ -163,7 +163,7 @@ - (void)sync_processRelationshipsUsingDictionary:(NSDictionary *)objectDict
         if (relationship.isToMany) {
             [self sync_processToManyRelationship:relationship usingDictionary:objectDict andParent:parent dataStack:dataStack];
         } else {
-            if (parent) {
+            if (parent && relationship.destinationEntity.name == parent.entity.name) {
```
Comment (NSElvis) [inline Source/Sync.m]: You might want to use `isEqualToString` instead of `==`

**Comment by NSElvis** [comment]
@kostiakoval Would you mind adding a test for this?

The common workflow is to add a bug that proves the failure, then fix the thing.

**Comment by kostiakoval** [comment]
:smile_cat:  I did it.
to test. just revert last change it Sync 
> if (parent && relationship.destinationEntity.name == parent.entity.name) {
> if (parent) {

And it will crash
@NSElvis

**Comment by NSElvis** [comment]
@kostiakoval Great work Konstantin!

**Pull request merged**